File: python/samples/action_submit_execute_sample.py
from fluent_cards import AdaptiveCardBuilder, TextSize, TextWeight


def create_action_submit_execute_card() -> dict:
    """Creates a card with Action.Execute and Action.Submit actions and custom verbs/data."""
    return (
        AdaptiveCardBuilder.create()
        .with_version('1.4')
        .add_text_block(
            lambda tb: tb.with_text('welcome to ac 11')
            .with_size(TextSize.Large)
            .with_weight(TextWeight.Bolder)
        )
        .add_text_block(lambda tb: tb.with_text('click the buttons below'))
        .add_action(
            lambda a: a.execute('Test AC Action')
            .with_data({'message': 'button clicked !!'})
            .with_verb('testAction')
        )
        .add_action(
            lambda a: a.submit('Open Task Module').with_data(
                {'msteams': {'type': 'task/fetch'}}
            )
        )
        .add_action(
            lambda a: a.execute('Request File Upload').with_verb('requestFileUpload')
        )
        .build()
    )

File: node/samples/actionSubmitExecuteSample.ts
import { AdaptiveCardBuilder, TextSize, TextWeight } from 'fluent-cards';

/** Creates a card with Action.Execute and Action.Submit actions and custom verbs/data. */
export function createActionSubmitExecuteCard() {
  return AdaptiveCardBuilder.create()
    .withVersion('1.4')
    .addTextBlock((tb) =>
      tb
        .withText('welcome to ac 11')
        .withSize(TextSize.Large)
        .withWeight(TextWeight.Bolder),
    )
    .addTextBlock((tb) => tb.withText('click the buttons below'))
    .addAction((a) =>
      a
        .execute('Test AC Action')
        .withData({ message: 'button clicked !!' })
        .withVerb('testAction'),
    )
    .addAction((a) =>
      a
        .submit('Open Task Module')
        .withData({ msteams: { type: 'task/fetch' } }),
    )
    .addAction((a) => a.execute('Request File Upload').withVerb('requestFileUpload'))
    .build();
}
